app/models/policy_version.rb

```diff
@@ -86,6 +86,61 @@ def before_save
     self.policy_sha256 = Digest::SHA256.hexdigest(policy_text)
   end
 
+  def after_save
+    ensure_versions_limit
+  rescue => e
+    Rails.logger.error("Error while ensuring policy '#{id}' versions retention limit: #{e.inspect}")
+  end
+
+  def ensure_versions_limit
+    log_versions_off_limit
+    delete_versions_off_limit
+  end
+
+  def log_versions_off_limit
```

Comment:
what's versions off limit? are these the versions that are beyond the defined limit? e.g version 21?

Reply:
also - what do we log for each one?

Reply:
Yes for both questions